src/lib/GSENSOR/stk8baxx.cpp

```diff
@@ -0,0 +1,265 @@
+#ifdef HAS_GSENSOR_STK8xxx
+#include "stk8baxx.h"
+#include "logging.h"
+
+#define PID_SIZE	16
+uint8_t chipid_temp = 0x00;
+uint8_t stk8xxx_pid_list[PID_SIZE] = {STK8xxx_CHIPID_VAL, STK8BA50_X_CHIPID_VAL, STK8BA5X_CHIPID_VAL, STK8327_CHIPID_VAL};
+
+void STK8xxx::ReadAccRegister(uint8_t reg, uint8_t *data)
+{
+    Wire.beginTransmission(STK8xxx_SLAVE_ADDRESS);
+    Wire.write(reg);
+    Wire.endTransmission();
+
+    Wire.requestFrom(STK8xxx_SLAVE_ADDRESS, 1);    // request 1 bytes from slave device
+    *data = Wire.read(); // receive a byte
+}
+
+void STK8xxx::WriteAccRegister(uint8_t reg, uint8_t data)
+{
+    Wire.beginTransmission(STK8xxx_SLAVE_ADDRESS);
+    Wire.write(reg);
+    Wire.write(data);
+    Wire.endTransmission();
+}
+
+/*
+ * Anymotion init register
+ * Must be placed after the software reset
+ *
+ */
+void STK8xxx::STK8xxx_Anymotion_init()
+{
+	unsigned char ARegAddr, ARegWriteValue;
+	
+	/* Enable X Y Z-axis any-motion (slope) interrupt */
+    ARegAddr       = STK8xxx_REG_INTEN1;
+    ARegWriteValue = 0x07;
+    WriteAccRegister(ARegAddr, ARegWriteValue);
+
+	/* Set anymotion Interrupt trigger threshold */
+    ARegAddr       = STK8xxx_REG_SLOPETHD;
+    ARegWriteValue = 0x14; 
```

Comment:
This is the default value and can be removed.  Or replaced with

#define STK8xxx_REG_SLOPETHD_DEFAULT 0x14